LazarevDV/lab_4/lab_4.cpp

```diff
@@ -0,0 +1,120 @@
+#include <iostream>
+#include <conio.h>
+
+
+using namespace std;
+
+float power() {
+    float number, counter = 1; //number число вводимое в степень, pow сама степень, counter счетчик
+    int pow;
+    cout << "Введите число и степень через пробел: ";
+    cin >> number >> pow;
+    if (pow < 0) {
+        for (int i = 0; i < abs(pow); i++) {
+            counter = counter / number;
+        } 
+    }
+    else {
+        for (int i = 0; i < abs(pow); i++) {
+            counter *= number;
+        }
+    }
+    if (abs(pow) == 3) {
+        cout << number << " В " << pow << "-ей степени равно: " << counter;
+    }
+    else {
+        cout << number << " В " << pow << "-ой степени равно: " << counter;
+    }
+    cout << endl;
+    return number;
+}
+
+int fact() {
+    int number, counter = 1; //number край факториала, counter счетчик
+    cout << "Введите число для нахождения факториала: ";
+    cin >> number;
+    for (int i = 1; i < number + 1; i++) {
+        counter *= i;
+    }
+    cout << "Факториал равен: " << counter;
+    cout << endl;
+    return number;
+}
+
+float square() {
+    float number, temp = 0, counter = 1; //number число для нахождения корня, counter счетчик
+    cout << "Введите число для его квадратного корня: ";
+    cin >> number;
+    if (number <= 0) {
+        if (number == 0) {
+            cout << "Корень нуля равен 0\n";
+        }
+        if (number < 0) {
+            cout << "Невозможно найти корень отрицательного числа, попробуйте еще раз!\n";
+        }
+    }
+    else {
+        while (counter != temp) {
+            temp = counter;
+            counter = (temp + number / temp) / 2.0;
+        }
+        cout << "Корень " << number << " равен: " << counter;
+        cout << endl;
+    }
+    return number;
+}
+
+int simplicity() {
+    int number; //number число для проверки на простоту 
+    bool flag; 
+    cout << "Введите число для проверки на простоту: ";
+    cin >> number;
+    if (number <= 0)
+        cout << "Вы ввели '0' или отрицательное число, попробуйте еще раз!\n"; 
+    else
+    {
+        flag = true;
+        for (int i = 2; i <= sqrt(number) + 1; i++) {
+            if (number % i == 0)
+                flag = false;
+        }
+        
+        if (number == 1)
+            flag = false;
+        if (number == 2)
+            flag = true;
+        if (flag)
+            cout << "Число " << number << " простое\n";
+        else
+            cout << "Число " << number << " не простое\n";
+    }
+    return 0;
```

Comment:
Function that check if value if prime or not should not print anything. It should return true in case of prime and false otherwise